lib/Reader/readerir.cpp

```diff
@@ -2020,6 +2020,91 @@ bool GenIR::isUnmanagedPointerType(llvm::Type *Type) {
   return Type->isPointerTy() && !isManagedPointerType(Type);
 }
 
+bool GenIR::isArrayType(llvm::Type *ArrayTy) {
```

Comment:
Should this also validate the presence of something that looks like a length field?